apisix/plugins/echo.lua

```diff
@@ -70,4 +102,23 @@ function _M.body_filter(conf, ctx)
     ngx.arg[2] = true
 end
 
+function _M.access(conf, ctx)
+    local auth_header = core.request.header(ctx, "Authorization")
```

Comment:
This `core.request.header` function returns the value of the specified header. So it should be like this:
```
local value = core.request.header(ctx,  conf.auth_header)
if value ~= conf.auth_value then
    return 401, "unauthorized request."
end
```
In addition, the boundary conditions need to be considered.

Reply:
@agile6v